### ISSUE TITLE
Refactoring odo in small chunks

### ISSUE THREAD

**Issue by dharmit** [openshift/odo#4298]
## What are we doing?
For Sprint 194, we will work on refactoring odo code in small chunks. What this means is that we will try to:
1. remove functions or code that's repeating across packages/layers but doing (almost) the same thing
1. break large function that's doing multiple things into smaller functions that are easier to read and understand
1. add more comments to the code to make it simpler for other contributors to understand
1. unify repeating code into functions
1. clean the code iteratively with small and focussed PRs

### Why are we not doing what's suggested/discussed in #4057? What about the refactoring issues like #4081, #4093, #4115, #4145, and more?
We will still be addressing these issues when we're more confident. Yes, we're not enough confident right now to fully embrace Go interfaces and make our code adhere to Go best practices. If you have any ideas/suggestions, or feel like helping out, please do! The goal of odo project still remains the same - to achieve code architecture similar to what's discussed in #4057.

### What's the desired outcome of this issue?
Parts of our code were added/developed by contributors who are not active at the moment. While working on this issue we would like to better understand the flow of things across the code instead of limiting ourselves only to the packages we contributed to in the past.

### What's the Acceptance Criteria for this issue?
There's no predefined criteria except the points mentioned under "What are we doing?" As we open & merge PRs, we will mention them in the comments on this issue.

**Comment by dharmit** [comment]
@kadel @mik-dass @adisky let me know if anything about issue description needs to be changed.

**Comment by dharmit** [comment]
In Sprint 194, as a part of this issue, I started stepping through the code to understand where we can make improvements. I ended up creating:
* PR 4322
* PR 4323
* Issue #4313

And to the best of my knowledge, @mik-dass opened PR 4311.

In Sprint 195, I intend to continue stepping through the code as much as I can. AFAIK, @mik-dass is going to open a PR to refactor storage bits once PR 4311 is merged.

@mik-dass correct me if I'm wrong.
